src/resolvers/dockerfileTemplate.ts

```diff
@@ -18,8 +32,13 @@ export class DockerfileTemplateResolver implements Resolver {
 		this.hasDockerfileTemplate = true;
 	}
 
-	public needsEntry(filepath: string) {
-		return basename(filepath) === 'Dockerfile.template';
+	public needsEntry(entry: TarEntryPath, specifiedDockerfilePath?: string) {
+		return (
+			entry.extension === '.template' &&
+			(specifiedDockerfilePath
+				? entry.path === specifiedDockerfilePath
+				: entry.pathNoExt === 'Dockerfile')
```

Comment:
Two cases:
* If a specifiedDockeriflePath was specified, say `'service/MyDockerfile.template'`, then this resolver (`DockerfileTemplateResolver `) will only match a tar entry whose full path is identical to that (`entry.path === specifiedDockerfilePath`), and provided that the specifiedDockeriflePath has a '.template' file extension.
* Otherwise, it will match `Dockerfile.template` at the root of the project directory tree, as `pathNoExt` is the full path (minus file extension) which must be exactly `'Dockerfile'`. This root-only rule was previously enforced in `resolveTarStreamOnEntry()` through a test: `if (path.normalize(name).split(path.sep).length === 1)`. That test had been added by a recent-ish pull request:
_fix: Correctly resolve contexts with multiple bundle descriptions_
https://github.com/balena-io-modules/resin-bundle-resolve/pull/32/files